src/Microsoft.ComponentDetection.Detectors/nuget/NuGetPackagesConfigDetector.cs

```diff
@@ -53,17 +53,27 @@ protected override Task OnFileFoundAsync(ProcessRequest processRequest, IDiction
     {
         try
         {
+            var singleFileComponentRecorder = processRequest.SingleFileComponentRecorder;
             var packagesConfig = new PackagesConfigReader(processRequest.ComponentStream.Stream);
             foreach (var package in packagesConfig.GetPackages(allowDuplicatePackageIds: true))
             {
-                processRequest.SingleFileComponentRecorder.RegisterUsage(
+                singleFileComponentRecorder.RegisterUsage(
                     new DetectedComponent(
                         new NuGetComponent(
                             package.PackageIdentity.Id,
                             package.PackageIdentity.Version.ToNormalizedString())),
                     true,
                     null,
+                    /* TODO: Is this really the same concept?
+                       Docs for NuGet say packages.config development dependencies are just not persisted as dependencies in the package.
+                       That is not same as excluding from the output directory / runtime. */
                     package.IsDevelopmentDependency);
+
+                // get the actual component in case it already exists
+                var libraryComponent = singleFileComponentRecorder.GetComponent(package.PackageIdentity.Id);
+
+                // Add framework information to the actual component
+                ((NuGetComponent)libraryComponent.Component).TargetFrameworks.Add(package.TargetFramework.GetShortFolderName());
```

Comment:
Are sure `package.TargetFramework` is always included? Might be worth adding a `string.IsNullOrWhiteSpace(package.TargetFramework?.GetShortFolderName())` and not add it to TargetFramework list if it isn't valid.

Reply:
Yeah, I can add that.   Agreed that we can't trust what packages.config exposes - it's user defined.  If we made this read the project it could do better, but I'm not inclined to do much for packages.config since it's not going to benefit from the auto-dismissal.